src/codegen/llvm.zig

```diff
@@ -11175,6 +11176,8 @@ fn backendSupportsF16(target: std.Target) bool {
 fn backendSupportsF128(target: std.Target) bool {
     return switch (target.cpu.arch) {
         .amdgcn => false,
+        .x86_64, .x86 => !std.Target.x86.featureSetHas(target.cpu.features, .soft_float),
```

Comment:
Is soft_float support relevant for f16 as well? I see that this line already exist for f80.

Reply:
Probably. This is the only problem we hit currently though, but https://github.com/ziglang/zig/issues/15939 is for cleaning this up in the future.